src/background/in-dapp-notifications.ts

```diff
@@ -0,0 +1,84 @@
+import browser from 'webextension-polyfill';
+import { networksStore } from 'src/modules/networks/networks-store.background';
+import type { ChainId } from 'src/modules/ethereum/transactions/ChainId';
+import type { Chain } from 'src/modules/networks/Chain';
+import type { InDappNotification } from 'src/shared/types/InDappNotification';
+import { emitter } from './events';
+
+declare global {
+  interface Window {
+    showNotification(notification: InDappNotification): void;
+  }
+}
+
+async function getActiveTabIdWithOrigin(origin: string) {
+  // In dev environment the tab we're looking for doesn't have to be active or in the current window.
+  // Otherwise it would be hard to debug the chainSwitchError case
+  const query =
+    process.env.NODE_ENV === 'production'
+      ? { currentWindow: true, active: true }
+      : {};
```

Comment:
Why does tab needs to be active at all?

Reply:
I actually thought about it. In one of the previous iterations, the notification appeared on all tabs matching the given origin. But then I wondered: why do we want to show notifications on other tabs in the first place?

Reply:
Why not? I think it's relevant. User can also have dapp open in a separate tiled window. Let's discuss!

Reply:
Let me test both options and reconsider. It might indeed be an artificial limitation

Reply:
The only downside is that when a tab in another window is inactive, the notification could be displayed like this. I mean that when the tab is activated, the notification's hide animation might already be in progress, which can look odd

Uploading Screen Recording 2024-06-18 at 17.42.44.mov…